webapp/src/editortoolbar.tsx

```diff
@@ -225,6 +225,13 @@ export class EditorToolbar extends data.Component<ISettingsProps, EditorToolbarS
             case "notimpl":
                 modalBody = lf("WebUSB is not supported by this browser; please check for updates.");
                 break;
+            default:
```

Comment:
would

```typescript
        (this.props.parent as ProjectView).chooseHwDialog.show();
```

(similar to `onHwDownloadClick` above) work here?

Reply:
I just played with this and it doesn't really give us the desired behavior. If I just show the hardware dialog and then return, avoiding the "Cannot Connect Device" dialog, when you select the hardware, it automatically downloads a file version of your program which takes a while instead of prompting the user for usb stuff. 

I can play with this option a bit to see what I can come up with.

Reply:
Another thing I worry about is that showing the choose hardware option off the bat seems buggy and that choosing the different download options will just do the same thing.

Reply:
Isn't that basically what your instructions are telling them to do?

Reply:
ah so arcade does nothave target flag set to have webusb as default experience, so it fallbacks to download as first experience, makes sense. Could add another flag by following path `skipDownload` flag goes starting in `app.tsx`, and have that initiate pair on completion instead of compile / download, though that may make this one take a bit longer since the path is a lil long

Reply:
Oh, using `this.onHwItemClick()` looks like it works. The important part is that skipping the download is set to true, which it is in that function. With using the `{...}.hwDialog.show()` method, we use the default of `skipDownload` which is false, which does the compile and file download.

Reply:
The one thing that is kinda a downside of using the `onHwItemClick` route is that you get a very different experience after you've connected and disconnected a device.
If y'all could play with it and see what you think, that would be super helpful: https://arcade.makecode.com/app/2cf11bbee78e6da5c33395047ede2b010faaccec-e3d341ab60#.

Steps:
1. Connect to a WebUsb device under the connect device option (get the hardware dialog)
2. Disconnect that device
3. Select connect device again

Reply:
You can also compare the experience with this upload target with the one in the PR comment and see what's better.

Reply:
The experiences after disconnecting seemed the same to me. Maybe I'm missing something?

But I'd say I much prefer the experience where it automatically pops up the "choose device" dialog. Saves me 3 steps (reading the instructions, going to click the "choose hardware" manually, then going back and re-clicking the "connect device" again). Handling that automatically and just directly asking which device I have feels much more intuitive.

Reply:
Yeah, I'm more thinking of the fact that when you click the connect device the first time, you get the dialog to select hardware. Once you disconnect and reconnect, you then only get the modal that tells you to connect via webusb.

I also had an experience yesterday that I was playing with that triggered the pair dialog after the hardware was chosen and I thought it might be a bit weird to have to choose hardware in one case and not the other. That experience was pretty buggy, though, and I think I'd want to leave it with just opening the hardware dialog if hardware hasn't been chosen yet.